docs/dev-tools/extensions/EXTRACT_STYLING.md

```diff
@@ -0,0 +1,79 @@
+# Otter Extract Styling to o3r-var
+
+Add commands to extract scss properties to Otter variable
+
+## Features
+
+### Extract a single variable
+
+![](../../../.attachments/dev-tools/vscode-extensions/gif/extract-to-o3r-var.gif)
+
+### Extract all variables
+
+![](../../../.attachments/dev-tools/vscode-extensions/gif/extract-all-to-o3r-var.gif)
+
+### Commands
+
+- `Otter: Extract SCSS property to o3r.variable` : extract a single pair propery/value into to an otter variable
+
+> must be on the line on which you want to perform the command
+
+- `Otter: Extract all SCSS properties to o3r.variable` : extract all the matches properties/values into otter variables
+
+### Extension Settings
+
+This extension contributes the following settings:
+
+- `otter.extract.styling.forbiddenWords`: Array of words to exclude from the final result of a generated variable
+- `otter.extract.styling.prefix`: If not empty, will add a prefix to all the generated variables.
+- `otter.extract.styling.functionImportPath`: Specify the import to add for the Otter function 'o3r.variable'.
+
+## Example
+
+Before
+
+```scss
+.app-header {
+  .title-text {
+    color: red;
+  }
+}
+```
+
+Use the command `Otter: Extract SCSS property to o3r-var` where the cursor is set
+> must be on the line on which you want to perform the command
+
+After
+
+```scss
+@use "@o3r/styling" as o3r;
```

Comment:
Maybe we should generate this part in the theme.scss file where it's supposed to be.